### PR TITLE
New product documentation

### PR DESCRIPTION
**Pull request by ggbecker**
Introduce documentation on how to create a new product in the project.

Also contains simple bash script to help creating the directory structure and some of the needed files.

Relates to #4272

### CODE REVIEW

File: example/README.md [hunk 1]
```diff
@@ -10,11 +10,49 @@ This product was introduced to show a few aspects:
...
 This will build only the example profile; this can then be used
 with [OpenSCAP](https://github.com/OpenSCAP/openscap) or
 [scap-workbench](https://github.com/OpenSCAP/scap-workbench) for
 testing purposes.
+
+## Creating a new Product
+
+Follow these steps to create a new product in the project:
+
+1. Create a new folder in the root directory which will holds the files related to your new product. To illustrate the process we will use the name `custom6` which basically means that the product is called `custom` and the major version is `6`.
```
Comment (jan-cerny) [inline example/README.md]: holds -> hold

File: example/README.md [hunk 2]
```diff
@@ -10,11 +10,49 @@ This product was introduced to show a few aspects:
...
+message(STATUS "Custom 6: ${SSG_PRODUCT_CUSTOM6}")
+message(STATUS "JBoss EAP 6: ${SSG_PRODUCT_EAP6}")
+...
+```
+```
+...
+if (SSG_PRODUCT_DEBIAN8)
+    add_subdirectory("debian8")
+endif()
+if (SSG_PRODUCT_CUSTOM6
```
Comment (jan-cerny) [inline example/README.md]: missing )

File: example/README.md [hunk 3]
```diff
@@ -10,11 +10,49 @@ This product was introduced to show a few aspects:
...
+      add_subdirectory("custom6")
+endif()
+if (SSG_PRODUCT_EAP6)
+    add_subdirectory("eap6")
+endif()
+...
+```
```
Comment (jan-cerny) [inline example/README.md]: they also should add it in `./build_product`

File: example/README.md [hunk 4]
```diff
@@ -10,11 +10,49 @@ This product was introduced to show a few aspects:
...
+1. Create a new folder in the root directory which will holds the files related to your new product. To illustrate the process we will use the name `custom6` which basically means that the product is called `custom` and the major version is `6`.
+```
+cd content
+mkdir custom6
+```
+2. Add the product to [CMakeLists.txt](../CMakeLists.txt) by adding the following lines:
+```
+...
+option(SSG_PRODUCT_DEBIAN8 "If enabled, the Debian 8 SCAP content will be built" TRUE)
+option(SSG_PRODUCT_CUSTOM6 "If enabled, the Custom 6 SCAP content will be built" TRUE)
```
Comment (jan-cerny) [inline example/README.md]: I suggest doing this in a "diff"-like format so that it will obvious they will add only the line with custom

Comment (ggbecker) [reply example/README.md]: I have changed to make the addition bold instead. What do you think about?

I didn't find an easy way to make it "diff"-like, do have any suggestions? I would find painful to put minus (-) and plus (+) manually.

File: example/README.md (hunk 3 above)
Comment (ggbecker) [reply example/README.md]: I'll add this instruction as well. Thanks.

File: example/README.md (hunk 2 above)
Comment (ggbecker) [reply example/README.md]: fixed

File: example/README.md (hunk 1 above)
Comment (ggbecker) [reply example/README.md]: fixed

File: example/README.md (hunk 4 above)
Comment (ggbecker) [reply example/README.md]: Assuming that setting text to `bold` is enough. Resolving conversation.

File: example/README.md (hunk 3 above)
Comment (ggbecker) [reply example/README.md]: Added.

**Comment by ggbecker** [comment]
I have addressed review requests. But the documentation is still not complete. Further work needs to be done.

**Comment by ggbecker** [comment]
Added four more items to the documentation. Still in progress.

**Comment by ggbecker** [comment]
More sections added. I looks that only few items are missing now, mainly in the `ssgconstants.py` file.

**Comment by ggbecker** [comment]
- Added remaining sections for xslt transformation files
- Added script which create every file needed for a new product but files which need manual editing.

It is ready to review.

Note: the script is there in a best effort case, it lacks documentation on how to use it but it might help. Maybe it can evolve to something more useful.

**Comment by jan-cerny** [comment]
@ggbecker This is a great contribution!

**Pull request merged**